app/domain/authentication/authn_jwt/orchestrate_authentication.rb

```diff
@@ -0,0 +1,47 @@
+require 'command_class'
+
+# This class is the starting point of the JWT authenticate requests, responsible to identify the vendor configuration and to run the JWT authenticator
+module Authentication
+  module AuthnJwt
+
+    OrchestrateAuthentication ||= CommandClass.new(
+      dependencies: {
+        validate_uri_based_parameters: Authentication::AuthnJwt::InputValidation::ValidateUriBasedParameters.new,
+        create_vendor_configuration: Authentication::AuthnJwt::VendorConfigurations::CreateVendorConfiguration.new,
+        jwt_authenticator: Authentication::AuthnJwt::Authenticator.new,
+        logger: Rails.logger
+      },
+      inputs: %i[authenticator_input enabled_authenticators]
+    ) do
+
+      def call
+        validate_uri_based_parameters
+        authenticate_jwt
+      end
+
+      private
+
+      def validate_uri_based_parameters
+        @validate_uri_based_parameters.call(
+          authenticator_input: @authenticator_input,
+          enabled_authenticators: @enabled_authenticators
+        )
+      end
+
+      def authenticate_jwt
+        @logger.info(LogMessages::Authentication::AuthnJwt::JwtAuthenticatorEntryPoint.new(@authenticator_input.authenticator_name))
+
+        @jwt_authenticator.call(
+          jwt_configuration: jwt_authenticator_configuration,
+          authenticator_input: @authenticator_input
+        )
+      end
+
+      def jwt_authenticator_configuration
+        @jwt_authenticator_configuration ||= @create_vendor_configuration.call(
+          authenticator_input: @authenticator_input
+        )
+      end
```

Comment:
why is this done here? can't we do this inside `Authentication::AuthnJwt::Authenticator`? this way the authenticator object will just get the `authenticator_input` as input and `Authentication::AuthnJwt::VendorConfigurations::CreateVendorConfiguration.new` as a dependency. what's the value in creating it here only to pass it along?

Reply:
i actually think that `validate_uri_based_parameters` can also be done in `Authentication::AuthnJwt::Authenticator` (and you already get where i'm going to :) )

Reply:
lets talk about it